about.go

```diff
@@ -118,7 +148,56 @@ func getAboutCustomDataFieldValues(aboutConfigMap map[string]interface{}, aboutF
 	return mapValue
 }
 
-func About(statusEndpoints []StatusEndpoint, protocol string, aboutFilePath string, versionFilePath string, customData map[string]interface{}) string {
+func statusEndpointsToDependencies(statusEndpoints []StatusEndpoint) []Dependency {
+	dependencies := make([]Dependency, len(statusEndpoints))
+	for _, statusEndpoint := range statusEndpoints {
+		dependencies = append(dependencies, Dependency{
+			Name:          statusEndpoint.Name,
+			StatusPath:    statusEndpoint.Slug,
+			Type:          statusEndpoint.Type,
+			IsTraversable: statusEndpoint.IsTraversable,
+		})
+	}
+	return dependencies
+}
+
+func statusEndpointsToDependenciesV2(statusEndpoints []StatusEndpoint) []Dependency {
+	dependencies := make([]Dependency, len(statusEndpoints))
+	for _, statusEndpoint := range statusEndpoints {
+		dependencies = append(dependencies, Dependency{
+			Name:          statusEndpoint.Name,
+			StatusPath:    statusEndpoint.Slug,
+			Type:          statusEndpoint.Type,
+			IsTraversable: statusEndpoint.IsTraversable,
+		})
+	}
+	return dependencies
+}
+
+func About(
+	statusEndpoints []StatusEndpoint,
+	protocol string, aboutFilePath string,
+	versionFilePath string,
+	customData map[string]interface{},
+	apiVersion int,
+	checkStatus bool,
+) string {
+	switch apiVersion {
+	case APIV1:
+		return aboutV1(statusEndpoints, protocol, aboutFilePath, versionFilePath, customData)
+	case APIV2:
+		return aboutV2(statusEndpoints, protocol, aboutFilePath, versionFilePath, customData, checkStatus)
+	}
+	// should never reach here
+	return ""
```

Comment:
I think we should 400

Reply:
actually, how much do we call about this ? because it would fail in http.go before this is called. so maybe blank string as the default is fine when version is not valid ?